test-adapter/_testadapter/tink_unittest/Injector.hx

```diff
@@ -7,29 +7,26 @@ import haxe.macro.Expr;
 using _testadapter.PatchTools;
 
 class Injector {
+	public static function buildCase():Array<Field> {
+		var fields = Context.getBuildFields();
+
+		for (field in fields) {
+			switch (field.name) {
+				case "shouldRun":
+					field.patch(Start, macro if (!_testadapter.data.TestFilter.shouldRunTest($v{Macro.filters}, this.suite.info.name, this.info.name))
```

Comment:
The reason for `testResults.positions.resolveClassName` is that `suite.name` only contains a class name and no package information. 
Since class names might no be unique we need a full class name that includes package.

If you open the tink_unittest sample project and run test adapter you will see no `Show source` button, codelens or gutter icon for `TestCase3` in test explorer, you will see it for `TestCase` and `TestCase2` because they don't have a package.